dts/bindings/input/gpio-keys.yaml

```diff
@@ -35,15 +35,23 @@ properties:
        Debouncing interval time in milliseconds.
        If not specified defaults to 30.
 
+  no-interrupts:
+    type: boolean
+    description: |
+      Do not use interrupts for the key GPIOs, poll the pin periodically at the
+      specified debounce-interval-ms instead.
```

Comment:
apologies if this is a stupid question but shouldn't there be an actual *polling* interval (shorter than debounce interval), otherwise it's not really "debouncing", or is it?

Reply:
If you really wanted to you could come up with a logic with fast polling and slow debouncing, but I don't really see the point in doing that, at that point you can just poll at the slower rate and obtain pretty much the same debouncing effect, and also avoid waking up the cpu at a fast rate.

Give this a try, I'm happy to rework this once again if turns out to be glitchy.

Reply:
Had a good look at the Linux polled driver as well, which does implement separate poll and debounce interval, and yeah generally it's used with a poll interval of 100ms and debounce interval of 5ms, which results in effectively the debounce interval being unused, unless I missed something.

Reply:
Good comment though.